lib/transformations.js

```diff
@@ -1,38 +1,65 @@
 'use strict'
+function vertical (matrix) {
+  return matrix.slice().reverse()
+}
 
- exports.flip = function (matrix, direction) {
-  if (!['none', 'horizontal', 'vertical', 'both'].includes(direction)) {
-    throw new Error('You can only rotate by 0, 90, 180, or 270 degrees')
-  }
-
-  function vertical (matrix) {
-    return matrix.reverse()
-  }
+function horizontal (matrix) {
+  return matrix.map(r => r.reverse() )
+}
 
-  function horizontal (matrix) {
-    return matrix.map(r => r.reverse() )
+function flip (matrix, direction) {
+  if (!['none', 'horizontal', 'vertical', 'both'].includes(direction)) {
+    throw new Error('You can only flip by none, horizontal, vertical, or both')
   }
 
-  if ( direction === 'none' ) { return matrix }
-  if ( direction === 'vertical' ) { return vertical(matrix) }
-  if ( direction === 'horizontal' ) { return horizontal(matrix) }
-  if ( direction === 'both' ) { return vertical(horizontal(matrix)) }
+  if ( direction === 'vertical' ) { matrix = vertical(matrix) }
+  else if ( direction === 'horizontal' ) { matrix = horizontal(matrix) }
+  else if ( direction === 'both' ) { matrix = vertical(horizontal(matrix)) }
+  return matrix;
 }
 
-exports.rotate = function (matrix, degrees) {
-  if (![0, 90, 180, 270].includes(degrees)) {
-    throw new Error('You can only rotate by 0, 90, 180, or 270 degrees')
-  }
+function transpose(matrix) {
+    for (let i = 0; i < matrix.length; i++) {
+        for (let j = 0; j < i; j++) {
+            let temp = matrix[i][j]
+            matrix[i][j] = matrix[j][i]
+            matrix[j][i] = temp
+        }
+    }
+    return matrix;
+}
 
- const rotated = matrix.map(r => r.slice()).reverse()
- 
- for (let i = 0; i < rotated.length; i++) {
-   for (let j = 0; j < i; j++) {
-     let temp = rotated[i][j]
-     rotated[i][j] = rotated[j][i]
-     rotated[j][i] = temp
-   }
- }
+function rotate(matrix, degrees) {
+    if (![0, 90, 180, 270].includes(degrees)) {
+        throw new Error('You can only rotate by 0, 90, 180, or 270 degrees.')
+    }
+    // Original font matrices encoded with 90 degree rotation in clockwise direction.
+    let rotated;
+    switch(degrees) {
```

Comment:
you could use the same function as I've done with flipping above here - bearing in mind that a hash is indexed by strings so you would need to convert the parameter to a string first: `rotationFunctions(`${degrees}`)`

Reply:
I agree that a hash is indexed by strings.  However, my experience is that when you store a field into a JavaScript object, it first converts the key to a string.  So, to do this, I think the conversion would be automatic.  I will test.

Reply:
I was correct on my above comment regarding hash key type of string.  If you pass in an integer, it will be converted to a string prior to lookup in an Object.

Please review entire transformation.js file. I tested both rotation and flip from the examples.  I also added 2 new examples illustrating the glyphRotate/glyphFlip options of scroll.  Hoping you have a display with which to test.